Apps/DiceRoller/DiceRoller/ViewController.m

```diff
@@ -30,22 +52,289 @@ - (void)didReceiveMemoryWarning{
+}
+
+// Login to iSENSE
+- (void) login:(NSString *)email withPassword:(NSString *)pass {
+    
+    UIAlertView *spinnerDialog = [self getDispatchDialogWithMessage:@"Logging in..."];
+    [spinnerDialog show];
+    
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        
+        RPerson *currUser = [api createSessionWithEmail:email andPassword:pass];
+        
+        dispatch_async(dispatch_get_main_queue(), ^{
+            
+            if (currUser != nil) {
+                
+                [self.view makeWaffle:[NSString stringWithFormat:@"Login as %@ successful", email]
+                             duration:WAFFLE_LENGTH_SHORT
+                             position:WAFFLE_BOTTOM
+                                image:WAFFLE_CHECKMARK];
+                userEmail = email;
+                password = pass;
+            } else {
+                
+                [self.view makeWaffle:@"Login failed"
+                             duration:WAFFLE_LENGTH_SHORT
+                             position:WAFFLE_BOTTOM
+                                image:WAFFLE_RED_X];
+            }
+            [spinnerDialog dismissWithClickedButtonIndex:0 animated:YES];
+        });
+    });
+}
+
+// Default dispatch_async dialog with custom spinner
+- (UIAlertView *) getDispatchDialogWithMessage:(NSString *)dString {
+    
+    UIAlertView *message = [[UIAlertView alloc] initWithTitle:dString
+                                                      message:nil
+                                                     delegate:self
+                                            cancelButtonTitle:nil
+                                            otherButtonTitles:nil];
+    UIActivityIndicatorView *spinner = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleWhiteLarge];
+    spinner.center = CGPointMake(139.5, 75.5);
+    [message addSubview:spinner];
+    [spinner startAnimating];
+    return message;
+}
+
+#pragma end - Credentials
+
+#pragma mark - Name
+
+- (IBAction)nameBtnOnClick:(id)sender {
+    
+    UIAlertView *enterNameAlart = [[UIAlertView alloc] initWithTitle:@"Enter a Data Set Name"
+                                                             message:@""
+                                                            delegate:self
+                                                   cancelButtonTitle:@"Cancel"
+                                                   otherButtonTitles:@"OK", nil];
+    [enterNameAlart setAlertViewStyle:UIAlertViewStylePlainTextInput];
+    enterNameAlart.tag = kNAME_DIALOG_TAG;
+    [enterNameAlart textFieldAtIndex:0].delegate = self;
+    [[enterNameAlart textFieldAtIndex:0] becomeFirstResponder];
+    [enterNameAlart textFieldAtIndex:0].placeholder = @"data set name";
+    
+    [enterNameAlart show];
+}
+
+#pragma end - Name
+
+-(void) uploadDatadie1:(int)num1 die2: (int)num2 sumOfDies: (int)sumNum numOfTests: (int)numTest{
+    
```

Comment:
Can we adjust the spacing of this? i.e. Remove the spaces between the `:` and `(int)`s:
`-(void) uploadDatadie1:(int)num1 die2:(int)num2 sumOfDies:(int)sumNum numOfTests:(int)numTest`